zenko-2018-07-08-json.api.json

```diff
@@ -0,0 +1,379 @@
+{
+    "version": "1.0",
+    "metadata": {
+        "apiVersion": "2018-07-08",
+        "endpointPrefix": "s3",
+        "protocol": "rest-json",
+        "serviceAbbreviation": "zenko",
+        "serviceFullName": "Zenko Cloudserver Client",
+        "signatureVersion": "v4",
+        "timestampFormat": "rfc822",
+        "uid": "zenko-2018-07-08-json"
+    },
+    "operations": {
+        "checkConnection": {
+            "http": {
+                "method": "GET",
+                "requestUri": "/_/backbeat/api/healthcheck"
+            }
+        },
+        "getLocationsStatus": {
+            "http": {
+                "method": "GET",
+                "requestUri": "/_/backbeat/api/crr/status"
+            },
+            "output": {
+                "type": "map",
+                "key": {},
+                "value": {}
+            }
+        },
+        "listFailed": {
+            "http": {
+                "method": "GET",
+                "requestUri": "/_/backbeat/api/crr/failed"
+            },
+            "input": {
+                "type": "structure",
+                "members": {
+                    "Marker": {
+                        "location": "querystring",
+                        "locationName": "marker"
+                    },
+                    "Sitename": {
+                        "location": "querystring",
+                        "locationName": "sitename"
+                    }
+                }
+            },
+            "output": {
+                "type": "structure",
+                "members": {
+                    "IsTruncated": {
+                        "type": "boolean"
+                    },
+                    "NextMarker": {
+                        "type": "string"
+                    },
+                    "Versions": {
+                        "type": "list",
+                        "member": {
+                            "shape": "responseShape"
+                        }
+                    }
+                }
+            }
+        },
+        "getFailedObject": {
```

Comment:
I had gone with `getFailedObject` instead of `listFailedObject` because  `list` sounds like a retrieval of `1+` objects.

Reply:
It's definitely an art to come up with good names 😄. I'm thinking `list` could still work since in a one-to-many context the object may fail at more than one site. Hence the response is still an array of objects.

Reply:
I will let @alexanderchan-scality to follow up with the change he needs as it requires changing pensieve-front project as well. It can be done when it's switched over.

Reply:
It can return multiple entries per object e.g. one entry for each site that the failure occurred. So if a version failed at multiple destinations, the response would be a listing multiple objects with those sites included as properties. For example:

```
    {
        IsTruncated: false,
        Versions: [{
            Bucket: <bucket>,
            Key: <key>,
            VersionId: <version-id>,
            StorageClass: 'awsbackend',
            Size: <size>,
            LastModified: <last-modified>,
        },
        {
            Bucket: <bucket>,
            Key: <key>,
            VersionId: <version-id>,
            StorageClass:  'gcpbackend',
            Size: <size>,
            LastModified: <last-modified>,
        }]
    }
```

I like the specificity of `listFailedObjectVersions` instead of `listFailedObjects`.